docs/performance.asciidoc

```diff
@@ -0,0 +1,153 @@
+[[performance-tuning]]
+
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/nodejs/current/performance.html[elastic.co]
+endif::[]
+
+== Performance Tuning
+
+There are many options available to tune agent performance.
+Which option to adjust depends on whether you are optimizing for speed,
+memory usage,
+bandwidth,
+or storage.
+
+[float]
+[[performance-sampling]]
+=== Sampling
+
+The first knob to reach for when tuning the performance of the agent is <<transaction-sample-rate,`transactionSampleRate`>>.
+Adjusting the sampling rate controls what percent of requests are traced.
+By default,
+the sample rate is set at `1.0`,
+meaning _all_ requests are traced.
+
+The sample rate will impact all four performance categories,
+so simply turning down the sample rate is an easy way to improve performance.
+
+[source,js]
+----
+require('elastic-apm-node').start({
+  transactionSampleRate: 0.2
+})
+----
+
+[float]
+[[performance-transaction-queue]]
+=== Transaction Queue
+
+The agent uses a queue to send to the apm server in batches.
+This queue uses both a time interval and a size limit to balance memory usage against socket descriptor saturation.
+
+[float]
+[[performance-max-queue-size]]
+==== Max Queue Size
+
+The <<max-queue-size,`maxQueueSize`>> controls the maximum number of request transactions that may remain in the queue before they must be sent.
```

Comment:
s/request transactions/transactions/

(all transactions are added to the queue, not only request transactions)